Thunder/server/stream_routes.py

```
@@ -1,303 +1,217 @@
-    loop = asyncio.get_running_loop()
-    try:
-        while True:
-            try:
-                yield await loop.run_in_executor(executor, next, sync_gen)
-            except StopIteration:
-                break
+            work_loads[client_id] -= 1
+            raise
+        
+    except (InvalidHash, FileNotFound) as e:
+        raise web.HTTPNotFound(text=str(e))
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)